### PR TITLE
Add vmlinux-arm64.h

### LINKED ISSUE #453: [Bug Report] Unable to compile trace/trace.go for arm64
### Checks

- [X] I have searched the existing issues
- [X] I have read the documentation
- [X] Is it your first time sumbitting an issue

### Current Behavior

#435 The implemented dae trace functionality seems unable to achieve cross-platform compilation. For example, it cannot compile packages for the arm64 architecture on amd64, and even fails to compile arm64 packages on arm64.

I found that the arm64 package compiled in GitHub Actions also encounters errors when compiling trace/trace.go, but it is simply skipped.

### Expected Behavior

Is this the intended behavior? However, according to the instructions in #435, it should support the arm64 architecture.

### Steps to Reproduce

`GOARCH=arm64 make`

### Environment

- **Dae version (use `dae --version`)**: the latest main branch
- **OS (e.g `cat /etc/os-release`)**: x86_64 linux
- **Kernel (e.g. `uname -a`)**: 6.7.2
- **Others**:

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jschwinger233**
This PR basically created a new file `vmlinux-arm64.h`, ran `mv vmlinux.h vmlinux-x86.h`, and let vmlinux.h choose header according to arch.

```
[gray/arm 37454f1] Add vmlinux-arm64.h
 3 files changed, 97295 insertions(+), 120799 deletions(-)
 create mode 100644 vmlinux-arm64.h
 copy vmlinux.h => vmlinux-x86.h (100%)
 rewrite vmlinux.h (99%)
```

To fix: https://github.com/daeuniverse/dae/issues/453

Github diff refuses to work, I'll paste new content in vmlinux.h below:

```c
#if defined(__TARGET_ARCH_x86)
#include "vmlinux-x86.h"
#elif defined(__TARGET_ARCH_arm64)
#include "vmlinux-arm64.h"
#else
/*
 * For other architectures, we don't have a vmlinux.h file. But the normal dae
 * bpf program doesn't need it. So we just include the x86 vmlinux.h file to
 * make the build pass.
 */
#include "vmlinux-x86.h"
#endif

```

### CODE REVIEW

**Comment by jschwinger233** [comment]
@mzz2017

**Comment by jschwinger233** [comment]
Hold on...

File: vmlinux.h [hunk 1]
```diff
@@ -3,5 +3,10 @@
 #elif defined(__TARGET_ARCH_arm64)
 #include "vmlinux-arm64.h"
 #else
-#error "Unknown architecture"
+/*
+ * For other architectures, we don't have a vmlinux.h file. But the normal dae
+ * bpf program doesn't need it. So we just include the x86 vmlinux.h file to
+ * make the build pass.
+ */
+#include "vmlinux-x86.h"
```
Comment (jschwinger233) [inline vmlinux.h:11]: Add a second patch to tolerate unknown arch.

**Comment by douglarek** [comment]
No rush intended, but this pull request has been open for over 3 weeks. Is there any concern preventing it from being merged?

**Comment by mzz2017** [comment]
@jschwinger233 可以出个 dae 的 pr 让测起来了

**Pull request merged**